code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -139,6 +142,13 @@
 		if(XENO_UPGRADE_THREE)
 			return XENO_UPGRADE_TWO
 
+/mob/living/carbon/xenomorph/proc/setup_job()
+	var/datum/job/xenomorph/xeno_job = SSjob.type_occupations[xeno_caste.job_type]
+	if(!xeno_job)
+		stack_trace("Unemployment has reached to a xeno, who has failed to become a [xeno_caste.job_type]")
+	apply_assigned_role_to_spawn(xeno_job)
```

Comment:
Is this not going to fail trying to assign null?

Reply:
Added a CRASH to handle it